perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2763,6 +2866,10 @@ public void removeRelationMembers(PerunSession sess, Group resultGroup, List<Mem
 	@Override
 	public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operandGroup, boolean parentFlag) throws WrongReferenceAttributeValueException, WrongAttributeValueException, GroupNotExistsException, InternalErrorException, GroupRelationAlreadyExists, GroupRelationNotAllowed {
 
+		//Lock affected groups
+		PerunLocksUtils.lockGroupMembership(resultGroup);
+		PerunLocksUtils.lockGroupMembership(operandGroup);
```

Comment:
If I understand changes in this PR correctly, then groups in operations like "updateGroup" are locked in reverse order of their names. This code could be cause of deadlock. 

Imagine that method "updateGroup" wants to lock groups ABC in their names reverse order. It locks groups C, B and than this method is called with groups A and B where A is result group and B is operand group. It manages to lock A and then waits because "updateGroup" method locked group B. And method "updateGroup" is waiting because this method locked group A. I believe these two locks should be done as well in reverse order of given groups names.